### PR TITLE
[ci] Change image to ci-linux:production with clang13

### PR DESCRIPTION
**Pull request by alvicsam**
Pipeline with current image doesn't work

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -19,7 +19,8 @@ workflow:
     - if: $CI_COMMIT_BRANCH
 
 .docker-env:                       &docker-env
-  image:                           paritytech/ink-ci-linux:production
+  # Temporary image, remove when clang in "ink-ci-linux:production" is updated to 13
```
Comment (bkchr) [inline .gitlab-ci.yml:22]: Is that still true? Don't you actually just created this image for the future?

Comment (alvicsam) [reply .gitlab-ci.yml:22]: It depends on how quickly clang in ci image for `ink` would be updated. 
If it takes too much time I'll recreate this image with `production` tag and setup it's updates. So currently it is just a quick fix

**Review by radupopa2010 — APPROVED** [review]

File: .gitlab-ci.yml (hunk 1 above)
Comment (bkchr) [reply .gitlab-ci.yml:22]: Do you keep track of this?

Comment (alvicsam) [reply .gitlab-ci.yml:22]: I'll create the issue

**Pull request merged**